### ISSUE TITLE
Re-anchor/position Tooltip?

### ISSUE THREAD

**Issue by stewartcelani** [ReactTooltip/react-tooltip#245]
Hello there! One big use for ReactTooltip in a project I'm working on is to mouse over urls and have a tooltip show a description of the site via an embedly call. When the embedly call completes it renders a small picture + description of the website and messes up the position of the tooltip. As soon as you move the mouse inside the element the tooltip will right itself.

Here's whats going on: 
![reacttooltip](https://cloud.githubusercontent.com/assets/391792/21878013/1df711de-d8df-11e6-9b1c-dac3099531ab.gif)

I have a callback when the embed loads so I've tried messing around by triggering a custom 'updatetooltip' global event that I set up a listener for from 'bindListener' in index.js. It receives the events fine (console.logs working as expected etc) but calling either updateTooltip or updatePosition doesn't seem to do anything (well updatePosition throws a cannot read property 'currentTarget' of undefined). Also tried calling hideTooltip and then showTooltip but the show doesn't go through. ReactTooltip.rebuild() neither. Bit stumped.

Any ideas on how I could go about solving this would be hugely appreciated. Thanks for all your work on this library!

**Comment by huumanoid** [comment]
I've faced the same issue. I don't know can I be helpful here, but I have something important to mention.

First of all, probably hotfix for you (warning: it's **awful!**)
```js
setTimeout(() => this.refs.tooltip.updateTooltip(e), 1000);
```

The problem is how position calculated.
Position depends on size of tooltip, of course, and somehow size obtained while calculating position is incorrect.

To clarify what I mean, I propose the following explanation (test case, scenario).

Let's we have the following jsx:
```jsx
<ReactTooltip id="test" />
<p
  data-for="test"
  data-tip="{some long text here, probably HTML, but plain text is enough}"
>
{content}
</p>
```
Let's say our React page has been already rendered in- DOM. Therefore, there is a `<div />` corresponding to our `ReactTooltip`. However, this `<div />` is hidden (i.e. it hasn't `show` class).

Now it's time to trigger tooltip (i.e. move mouse pointer over `<p>`).
When we do so, somewhere deep under the hood, tooltip's position is calculated.
I don't remember the exact calculations, but let's assume the following pseudocode:
```jsx
tooltip = ref(<ReactTooltip id="test" />)
div = getDOMElementCorrespondsToTooltip(tooltip)
// size - some DOM API function (probably getClientRects, but I do not remember exactly)
tooltip.position = F(..., size(div), ...)
```
Let's denote `size(div)` from calculations above as `used_size`.
Let's assume that tooltip has been shown and it's at wrong position (in OP's case, it actually IS).

Let's calculate `size(div)` again (e.g. using dev's console) and denote it as `actual_size`.
It's better to calculate `actual_size` after a while from when the tooltip has been shown.

**The fact**: `used_size != actual_size`.

I've tried to build simple test to reproduce this issue, but unfortunately I could not.

Can't wait for the weekend to research more.
